ArduPlane/RC_Channel.cpp

```diff
@@ -78,6 +78,23 @@ void RC_Channel_Plane::do_aux_function_q_assist_state(AuxSwitchPos ch_flag)
     }
 }
 
+void RC_Channel_Plane::do_aux_function_soaring_3pos(AuxSwitchPos ch_flag)
+{
+#if SOARING_ENABLED == ENABLED
+    switch (ch_flag) {
+    case AuxSwitchPos::LOW:
+        plane.g2.soaring_controller.set_enable_ch_state(SoaringController::ActiveStatus::SOARING_DISABLED);
+        break;
+    case AuxSwitchPos::MIDDLE:
+        plane.g2.soaring_controller.set_enable_ch_state(SoaringController::ActiveStatus::MANUAL_MODE_CHANGE);
+        break;
+    case AuxSwitchPos::HIGH:
+        plane.g2.soaring_controller.set_enable_ch_state(SoaringController::ActiveStatus::AUTO_MODE_CHANGE);
+        break;
+    }
+#endif
```

Comment:
nitpick here for next time. At a glance it looks like this switch is doing a lot but it's just selecting the status. A more "obviously correct" method would be something like this
```
SoaringController::ActiveStatus activeState;
switch (ch_flag) {
    case AuxSwitchPos::LOW:
        activeState = SoaringController::ActiveStatus::SOARING_DISABLED;
        break;
    case AuxSwitchPos::MIDDLE:
        activeState = SoaringController::ActiveStatus::MANUAL_MODE_CHANGE;
        break;
    case AuxSwitchPos::HIGH:
        activeState = SoaringController::ActiveStatus::AUTO_MODE_CHANGE;
        break;
    default:
        // do not set the ActiveState
        return;
    }
plane.g2.soaring_controller.set_enable_ch_state(activeState);
```

Or have case default fallthrough into ::LOW depending on the behavior you want.
```
SoaringController::ActiveStatus activeState = SoaringController::ActiveStatus::SOARING_DISABLED;
switch (ch_flag) {
    default:
    case AuxSwitchPos::LOW:
        // disabled by default in variable init
        break;
    case AuxSwitchPos::MIDDLE:
        activeState = SoaringController::ActiveStatus::MANUAL_MODE_CHANGE;
        break;
    case AuxSwitchPos::HIGH:
        activeState = SoaringController::ActiveStatus::AUTO_MODE_CHANGE;
        break;
    }
plane.g2.soaring_controller.set_enable_ch_state(activeState);
```

Reply:
I concede

Reply:
Updated to have a single call into soaring_controller similar to the your first example